_posts/2024-12-17-release-0_9_0.md

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title:  "kroxylicious release 0.9.0"
+date:   2024-12-17 00:00:00 +0000
+author: "Sam Barker"
+author_url: "https://github.com/SamBarker"
+categories:  [releases, kroxylicious, kafka3.9]
+---
+
+The Kroxylicious project is very pleased to announce the [0.9.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.9.0) release of Kroxylicious. This release introduces support for Apache Kafka 3.9. 
+
+We generally expect Kroxylicious to be both forwards and backwards compatible across Kafka version just like the Apache Kafka client and broker. While we tested with pre-release versions Kafka 3.9 we discovered after it was released that we had missed a case in the protocol version negotiation. When the API Versions request version was updated we were unable to decode the response. In the 0.9.0 release we adopt the same behaviour as the Kafka broker and respond with API version 0 if we do not support the protocol version requested by the client - we have also expanded our test coverage to get earlier warnings of similar issues in the future. 
+
+This release also marks the introduction of our new connection handling sate machine at the core of the proxy. We have evolved state machine at the core of the proxy to separate out the state carried at various stages of the connection life cycle and provide stronger rules around transitions between states. This is still a work in progress as its implications are rather wide-ranging however with think it gives us a stronger core to base everything else around. This does for the moment limit our ability to offer authentication offload support, we do intend to restore this eventually (please do get in touch if this use case is important to you).
+
+There are many quality of life improvements in this release (in no particular order)
```

Comment:
maybe we should link to the release notes somewhere near here once they're available in github

Reply:
Yeah, thats a good point.

Reply:
Previously we have just put this line instead of listing all the changes
```suggestion
See the [Changelog](https://github.com/kroxylicious/kroxylicious/blob/main/CHANGELOG.md#080) for a list of changes and summary of Deprecations, Changes and Removals.
```

Reply:
I know but the old style always felt a bit bland and boring to me. So I wanted to call out the core changes in the announcement. 

I started with the idea of calling out the new contributors inline but ended up moving to a different list